src/Courses/CourseCreation/CourseEditPage.tsx

```diff
@@ -1,51 +1,239 @@
+
+            if (res?.status !== 201) {
+                console.error('Post unit failed.');
+                return;
+            }
+    
+            const newUnitId = res?.data.data.id;
+            
+            console.log(`Currying createUnit with ${newUnitId}`);
+            const createTopicForUnit = _.curry(createTopic)(_, newUnitId);
+            // WARNING: Why does this need to be cast as any, when this pattern works below?
+            await Promise.all(unit.topics.map((createTopicForUnit as any)));
+        };
+        
+        const createTopic = async (topic: NewCourseTopicObj, courseUnitContentId: number) => {
+            let newTopic = new NewCourseTopicObj(topic);
+            newTopic.courseUnitContentId = courseUnitContentId;
+            const newTopicFields = [ 
+                'courseUnitContentId', 'topicTypeId', 'name', 'startDate', 'endDate', 'deadDate', 'partialExtend',
+            ];
+            let postObject = _.pick(newTopic, newTopicFields);
+            console.log('Creating topic', postObject);
+            let res = await AxiosRequest.post('/courses/topic', postObject);
+            console.log(res);
+            let topicId = res.data?.data?.id;
+            
+            const createProblemForTopic = _.curry(createProblem)(_, topicId);
+            await Promise.all(newTopic.questions.map(createProblemForTopic));
+        };
+        
+        const createProblem = async (problem: ProblemObject, courseTopicContentId: number) => {
+            let newProblem = new ProblemObject(problem);
+            const newProblemFields = [
+                'problemNumber', 'webworkQuestionPath', 'courseTopicContentId', 'weight', 'maxAttempts', 'hidden', 'optional'
+            ];
+            let postObject: any = _.pick(newProblem, newProblemFields);
+            postObject.courseTopicContentId = courseTopicContentId;
+            console.log('Creating problem', postObject, ' from ', problem);
+            const res = await AxiosRequest.post('/courses/question', postObject);
+            console.log(res);
+        };
+
+        const createCourse = async (course: CourseObject) => {
+            // Not every field belongs in the request.
+            const newCourseFields = ['curriculum', 'name', 'code', 'start', 'end', 'sectionCode', 'semesterCode'];
+            let postObject = _.pick(course, newCourseFields);
+            postObject.code = 'TODO';
+            // TODO: Fix naming for route, should be 'templateId'.
+
+            if (!courseId || courseId == undefined) {
+                // TODO: Move this to the useEffect, navigate away if it fails?
+                return;
+            }
+
+            postObject.curriculumId = parseInt(courseId, 10);
+            console.log('Creating a new course', postObject);
+            return await AxiosRequest.post('/courses', postObject);
+        };
+
+        let res = await createCourse(course);
+        console.log(res);
+        if (res?.status !== 201) {
```

Comment:
I would just let axios do the error handling, by default it checks between [200, 300)